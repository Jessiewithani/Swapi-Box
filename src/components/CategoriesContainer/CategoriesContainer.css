h1 {
    color: white;
    font-size: 75px;
}

.bow {
    padding: 25px 90px;
}

.categoriesContainer {
    border-radius: 10px;
    display: grid;
    grid-template-columns: 210px 210px 210px 210px;
    grid-gap: 20px;
    height: 70%;
    justify-content: flex-end;
    overflow: scroll;
    padding: 50px;
}

.helloHeader {
    background-color: pink;
    border-radius: 10px;
    display: flex;
    height: 50px;
}

h3 {
    font-size: 30px;
    margin: 12px;
}

.pinkbow {
    width: 300px;
    position: fixed;
    right: 30px;
    top: -15px;
}

.fav-heart {
    width: 20px;
    margin-right: 0px;
    
}

.pick-favorites {
    margin-right: 16.5em;
}