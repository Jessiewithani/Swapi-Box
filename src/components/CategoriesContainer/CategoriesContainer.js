import React from 'react';
import '../CategoriesContainer/CategoriesContainer.css';
import CategoryCard from '../CategoryCard/CategoryCard';
import PropTypes from 'prop-types';
import Bow from '../../images/Bow.png'
import Heart from '../../images/Heart.png'

const CategoriesContainer = ({ data, toggleFavorite, countFavorites }) => {
  const displayCards = data.map((card, index) => <CategoryCard 
    key={card.name + index}
    type={card.type}
    name={card.name}
    homeworld= {card.homeworld}
    species={card.species}
    population={card.population}
    model={card.model}
    vehicle_class={card.vehicle_class}
    passengers={card.passengers}
    terrain={card.terrain}
    climate={card.climate}
    residents={card.resident}
    toggleFavorite={toggleFavorite}
    />);

    const favoriteCount = countFavorites();

    return (
      <div className="bow">
          <img className="pinkbow" src={Bow} alt=""/>
        <h1>Hello Star Wars</h1>
        <header className="helloHeader">
          <h3><img className="fav-heart" src={Heart} alt= ""/> Faves: {favoriteCount}</h3>
          {(countFavorites() <1) && <h3 className="pick-favorites"><img className="fav-heart" src={Heart} alt= ""/> Select Your Favorites!</h3>}
          <p></p>
        </header>
        <article className= "categoriesContainer">
          {displayCards}
        </article>
      </div>
    )
}

export default CategoriesContainer;

CategoriesContainer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
  toggleFavorite: PropTypes.func,
  countFavorites: PropTypes.func
}