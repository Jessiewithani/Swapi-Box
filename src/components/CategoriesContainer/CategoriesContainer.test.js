import React from 'react';
import { shallow } from 'enzyme';
import CategoriesContainer from './CategoriesContainer';

describe('CategoriesContainer', () => {
  const mockPeopleData =[{name: 'Luke Skywalker'},{name: 'Darth Vader'}]
  const mockVehicleData =[{name: 'Sand Crawler'},{name: 'Snowspeeder'}]
  const mockPlanetData =[{name: 'Hoth'},{name: 'Endor'}]
  const mockFavoritesData =[{name: 'Hoth'}, {name: 'Luke Skywalker'}]
  
  it('should match snapshot', () => {
    const wrapper = shallow(<CategoriesContainer data={mockPeopleData}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<CategoriesContainer data={mockVehicleData}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<CategoriesContainer data={mockPlanetData}/>)
    expect(wrapper).toMatchSnapshot()
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<CategoriesContainer data={mockFavoritesData}/>)
    expect(wrapper).toMatchSnapshot()
  });

});