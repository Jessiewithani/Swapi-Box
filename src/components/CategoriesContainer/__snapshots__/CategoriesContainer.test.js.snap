// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesContainer should match snapshot 1`] = `
<div
  className="bow"
>
  <img
    alt=""
    className="pinkbow"
    src="Bow.png"
  />
  <h1>
    Hello Star Wars
  </h1>
  <header
    className="helloHeader"
  >
    <h3>
      <img
        alt=""
        className="fav-heart"
        src="Heart.png"
      />
       Faves: 
    </h3>
    <p />
  </header>
  <article
    className="categoriesContainer"
  >
    <CategoryCard
      key="Luke Skywalker0"
      name="Luke Skywalker"
    />
    <CategoryCard
      key="Darth Vader1"
      name="Darth Vader"
    />
  </article>
</div>
`;

exports[`CategoriesContainer should match snapshot 2`] = `
<div
  className="bow"
>
  <img
    alt=""
    className="pinkbow"
    src="Bow.png"
  />
  <h1>
    Hello Star Wars
  </h1>
  <header
    className="helloHeader"
  >
    <h3>
      <img
        alt=""
        className="fav-heart"
        src="Heart.png"
      />
       Faves: 
    </h3>
    <p />
  </header>
  <article
    className="categoriesContainer"
  >
    <CategoryCard
      key="Sand Crawler0"
      name="Sand Crawler"
    />
    <CategoryCard
      key="Snowspeeder1"
      name="Snowspeeder"
    />
  </article>
</div>
`;

exports[`CategoriesContainer should match snapshot 3`] = `
<div
  className="bow"
>
  <img
    alt=""
    className="pinkbow"
    src="Bow.png"
  />
  <h1>
    Hello Star Wars
  </h1>
  <header
    className="helloHeader"
  >
    <h3>
      <img
        alt=""
        className="fav-heart"
        src="Heart.png"
      />
       Faves: 
    </h3>
    <p />
  </header>
  <article
    className="categoriesContainer"
  >
    <CategoryCard
      key="Hoth0"
      name="Hoth"
    />
    <CategoryCard
      key="Endor1"
      name="Endor"
    />
  </article>
</div>
`;

exports[`CategoriesContainer should match snapshot 4`] = `
<div
  className="bow"
>
  <img
    alt=""
    className="pinkbow"
    src="Bow.png"
  />
  <h1>
    Hello Star Wars
  </h1>
  <header
    className="helloHeader"
  >
    <h3>
      <img
        alt=""
        className="fav-heart"
        src="Heart.png"
      />
       Faves: 
    </h3>
    <p />
  </header>
  <article
    className="categoriesContainer"
  >
    <CategoryCard
      key="Hoth0"
      name="Hoth"
    />
    <CategoryCard
      key="Luke Skywalker1"
      name="Luke Skywalker"
    />
  </article>
</div>
`;
